### PR TITLE
Changed lc.fold() phase argument to transit midpoint.

### LINKED ISSUE #357: Should `lc.fold()` take a `transit_midpoint` or `t0` argument?
Right now `LightCurve.fold()` takes the arguments `period` and `phase`.  During a heated debate :fire:, we wondered if it would be more useful to have it take `transit_midpoint` instead of `phase`, because `transit_midpoint` is more commonly available (e.g. in the NASA Exoplanet Archive it is available with parameter name `pl_tranmid`).

@christinahedges Would you be willing to prepare a PR?

### PR DESCRIPTION
**Pull request by christinahedges**
I've changed lc.fold() phase argument to transit midpoint to fix issue #357. I've also fixed #353 and fixed the unit test. 

### Fold now takes `transit_midpoint` as an argument, not phase.
image

Note that if `transit_midpoint` is in JD it will warn the user that it doesn't look right. It just raises a warning, not an exception, because it's conceivable that the user knows what they're doing. @barentsen do you want to chime in on this?

### Folded light curve attribute `time_original` is now in the correct order.
image

Fixes #357 
Fixes #353

### CODE REVIEW

File: lightkurve/lightcurve.py [hunk 1]
```diff
@@ -335,7 +335,7 @@ def flatten(self, window_length=101, polyorder=2, return_trend=False,
             return flatten_lc, trend_lc
         return flatten_lc
 
-    def fold(self, period, phase=0.):
+    def fold(self, period, transit_midpoint=0.):
         """Folds the lightcurve at a specified ``period`` and ``phase``.
```
Comment (barentsen) [inline lightkurve/lightcurve.py]:
```suggestion
        """Folds the lightcurve at a specified ``period`` and ``transit_midpoint``.
```

File: lightkurve/lightcurve.py [hunk 2]
```diff
@@ -347,15 +347,20 @@ def fold(self, period, phase=0.):
...
         Returns
         -------
         folded_lightcurve : LightCurve object
             A new ``LightCurve`` in which the data are folded and sorted by
             phase.
         """
+
+        if (transit_midpoint > 2450000) & ((self.time[0] < 2450000) | (self.time_format == 'bkjd')):
+            warnings.warn('Transit mid point appears to be in JD, however light curve time appears'
+                        ' to be in BKJD (i.e. JD - 2454833).', LightkurveWarning)
```
Comment (barentsen) [inline lightkurve/lightcurve.py]: Can you make this generic to work for `time_format == 'btjd'` as well?

**Comment by christinahedges** [comment]
@barentsen Travis failed because of remote data how do I restart it?

**Comment by barentsen** [comment]
When I login to Travis I see a button that says "Restart build".  Because you are a repository maintainer, maybe you do too?

**Comment by christinahedges** [comment]
Thanks for the nicer warning message @barentsen, merging now.

**Pull request merged**